up42/asset.py

```diff
@@ -166,3 +173,57 @@ def download(
 
         self.results = out_filepaths
         return out_filepaths
+
+    def download_stac_asset(
+        self, stac_asset: pystac.Asset, output_directory: Union[str, Path, None] = None
+    ) -> Path:
+        """
+        Downloads a STAC asset to a specified output directory.
+
+        Args:
+            self (object): The instance of the class calling this method.
+            stac_asset (pystac.Asset): The STAC asset to be downloaded.
+            output_directory (Union[str, Path, None], optional): The directory where the asset
+                will be downloaded. If not provided, a default directory structure will be used.
+
+        Returns:
+            Path: The path to the downloaded asset file.
+
+        Raises:
+            requests.exceptions.HTTPError: If there is an HTTP error during the download.
+
+        """
+        logger.info(f"Downloading STAC asset {stac_asset.title}")
+        # TODO: make sure each stac asset has title
+        if output_directory is None:
+            output_directory = (
+                Path.cwd()
+                / f"project_{self.auth.project_id}/asset_{self.asset_id}/{stac_asset.title}"
+            )
+        else:
+            output_directory = Path(output_directory)
+        output_directory.mkdir(parents=True, exist_ok=True)
+        logger.info(f"Download directory: {str(output_directory)}")
+        try:
+            up42_signed_token = self.auth._get_auth_token()
```

Comment:
Is it not possible to use the auth._request method. I think this would be better, instead to add headers and use our own request. Also, once we migrate to account authentication we will have to change it everywhere is used. Let me know what do you think.

Reply:
The asset.href endpoint returns the contents of the file.
That's why I decided to implement our own request, similar to how we do it in `download` [method](https://github.com/up42/up42-py/blob/320b11955d67ee985b079b4d82cc955b0ea328a0/up42/utils.py#L164).

Do you prefer to refactor the `auth._request` method instead?

Reply:
no it is OK, then lets use the token from the auth objects.